srv/modules/runners/ui_rgw.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+"""
+Consolidate any user interface rgw calls for Wolffish and openATTIC.
+
+All operations will happen using the rest-api of RadosGW.  The one execption
+is getting the credentials for an administrative user which is implemented
+here.
+"""
+
+import logging
+import os
+import json
+import re
+import glob
+import salt.client
+import salt.utils.minions
+
+log = logging.getLogger(__name__)
+
+
+class Radosgw(object):
+    """
+    Return a structure containing S3 keys and urls
+    """
+
+    def __init__(self, canned=None, cluster='ceph', pathname='/srv/salt/ceph/rgw/cache'):
+        """
+        Initialize and call routines
+        """
+        if canned:
+            self._canned(canned)
+        else:
+            self.cluster = cluster
+            self.credentials = {'access_key': None,
+                                'secret_key': None,
+                                'urls': []}
+            self.pathname = pathname
+            self._admin()
+            self._urls()
+
+    def _canned(self, canned):
+        """
+        Return examples for debugging without a working Ceph cluster
+        """
+        if canned == 1:
+            self.credentials = {'access_key': "ABCDEFGHIJKLMNOPQRST",
+                                'secret_key': "0123456789012345678901234567890123456789",
+                                'urls': ["http://rgw1"]}
+        elif canned == 2:
+            self.credentials = {'access_key': "ABCDEFGHIJKLMNOPQRST",
+                                'secret_key': "0123456789012345678901234567890123456789",
+                                'urls': ["http://red1",
+                                         "http://red2",
+                                         "http://blue1:8000",
+                                         "http://blue2:8000"]}
+
+    def _admin(self, filename="user.admin.json"):
+        """
+        Expect admin user file; otherwise, search for first system user.
+        Update access_key, secret_key
+        """
+        filepath = "{}/{}".format(self.pathname, filename)
+        if os.path.exists(filepath):
+            user = json.loads(open(filepath).read())
+        else:
+            user = None
+            for user_file in glob.glob("{}/user.*".format(self.pathname)):
+                user = json.loads(open(user_file).read())
+                if 'system' in user and user['system'] == "true":
+                    break
+                user = None
+            if not user:
+                # No system user
+                log.error("No system user for radosgw found")
+                return
```

Comment:
I don't know much about RGW credentials, so just return `success = True` if the contents of the credentials variable are OK to be used, and `success = False` otherwise.

@theanalyst I don't think it's possible from the runner to influence the HTTP return code of the salt-api.